### PR TITLE
Fix error on cleaning cache

### LINKED ISSUE #456: Allow reloading web view if cache deleted
If cache is not present then app should reload the story from the website. Currently is just shows all the stories but on opening it is blank. might be related to #448 maybe cache is the problem there too.

### PR DESCRIPTION
**Pull request by ShikherVerma**

### CODE REVIEW

**Comment by ShikherVerma** [comment]
I am unable to find where to `setRefreshing(false)`
Reload no cache is working fine.
However this still doesn't solve the issue. See #456 

ezgif com-video-to-gif

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java [hunk 1]
```diff
@@ -125,6 +127,16 @@ public void onProgressChanged(WebView view, int newProgress) {
             return false;
         });
         setWebViewSettings(mWebView.getSettings());
+        mSwipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipe_container);
+        mSwipeContainer.setColorSchemeResources(R.color.white);
+        mSwipeContainer.setProgressBackgroundColorSchemeResource(
+            AppUtils.getThemedResId(getActivity(), R.attr.colorAccent));
+        mSwipeContainer.setOnRefreshListener(() -> {
+            WebSettings webSettings = mWebView.getSettings();
+            webSettings.setCacheMode(WebSettings.LOAD_NO_CACHE);
```
Comment (hidroh) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: This should come after `setWebViewSettings()` which set cache mode to `LOAD_CACHE_ELSE_NETWORK`

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java [hunk 2]
```diff
@@ -125,6 +127,16 @@ public void onProgressChanged(WebView view, int newProgress) {
...
+            setWebViewSettings(webSettings);
+            mWebView.reload();
```
Comment (hidroh) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: Should call `WebView.loadUrl("about:blank")` first to clear resources if any from current page. See http://developer.android.com/reference/android/webkit/WebView.html#clearView()

**Comment by hidroh** [comment]
Upon pulling to refresh, instead of setting cache mode to `LOAD_NO_CACHE`, you can use `AppUtils.hasConnection()` to check for network connection and decide whether to use `LOAD_NO_CACHE` or `LOAD_CACHE_ONLY`.

File: app/src/main/res/layout/fragment_web.xml [hunk 3]
```diff
@@ -14,39 +14,44 @@
...
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    app:layout_behavior="@string/appbar_scrolling_view_behavior"
-    android:id="@+id/nested_scroll_view"
-    android:background="?attr/colorCardBackground"
-    android:scrollbarStyle="insideOverlay"
-    android:scrollbars="vertical"
+    android:id="@+id/swipe_container"
     android:layout_width="match_parent"
-    android:layout_height="wrap_content">
+    android:layout_height="match_parent"
```
Comment (hidroh) [inline app/src/main/res/layout/fragment_web.xml]: Can this be `wrap_content`? I remember last time I had to make it `wrap_content`, or else there are lots of empty spaces at the end of the page.

**Comment by hidroh** [comment]
Please don't forget to add tests for your changes as well.

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java (hunk 1 above)
Comment (ShikherVerma) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: you mean I should change
`setWebViewSettings(mWebView.getSettings());`
to
```
WebSettings webSettings = mWebView.getSettings();
webSettings.setCacheMode(WebSettings.LOAD_CACHE_ELSE_NETWORK);
setWebViewSettings(webSettings);
```

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java (hunk 2 above)
Comment (ShikherVerma) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: done

File: app/src/main/res/layout/fragment_web.xml (hunk 3 above)
Comment (ShikherVerma) [inline app/src/main/res/layout/fragment_web.xml]: done

**Comment by ShikherVerma** [comment]
@hidroh I am unable to find where to `setRefreshing(false)` to make SwipeRefreshLayout disappear.

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java (hunk 1 above)
Comment (hidroh) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: no just move `webSettings.setCacheMode(WebSettings.LOAD_NO_CACHE);` one line down, after `setWebViewSettings()`, as it sets cache mode inside the method.

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java [hunk 4]
```diff
@@ -196,6 +214,10 @@ private void bindContent() {
         }
     }
 
+    private void removeRefreshing() {
```
Comment (ShikherVerma) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: This never gets executed ! I thought ItemResponseListener would be called after the item is loaded.

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java (hunk 1 above)
Comment (ShikherVerma) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: done

File: app/src/main/java/io/github/hidroh/materialistic/WebFragment.java (hunk 4 above)
Comment (hidroh) [inline app/src/main/java/io/github/hidroh/materialistic/WebFragment.java]: Refreshing the webpage does not refresh the item, thus it's correct that this does not get executed. You should put this logic in `onProgressChanged` instead.

**Comment by hidroh** [comment]
@ShikherVerma there have been some changes to WebFragment since last time. I'll close your PR for now and manually merge your changes later. Thanks for contributing!

**Pull request closed without merging**